src/dict.c

```diff
@@ -1218,6 +1218,112 @@ char *stringFromLongLong(long long value) {
     return s;
 }
 
+/* Evolving dict into dictX. */
+dictX *dictEvolution(dict *d) {
+    dictX *dx = zmalloc(sizeof(*dx));
+
+    /* Copy the internal constants from dict to dictX. */
+    dx->type = d->type;
+    dx->rehashidx = 0;  /* when relocating elements using. */
+    dx->pauserehash = 0;
+    dx->ht_used[0] = 0;
+    signed char new_ht_size_exp = _dictNextExp(d->ht_used[0] + 1);
```

Comment:
Why pass `+ 1` to `_dictNextExp()` shouldn't the size (and therefor the `tableusednum`) be the same as in `d`?

Reply:
You are right, I will change this boundary treatment.